### ISSUE TITLE
Run-away CPU-hogging on MacOS

### ISSUE THREAD

**Issue by ppesek** [hluk/CopyQ#625]
For the second time now, I've had to manually kill off CopyQ because on an otherwise idle system, it was hogging up all my resources.  The context for this was a clipboard filled with a lot of short rich text cut/pastes (mailing address fragments) from Google Docs (Sheets) using the latest release of Firefox on the latest release of MacOS with the latest release of CopyQ (2.8.2).  

I wouldn't have known had the fan not started going crazy on my otherwise very quiet Mac Mini...

See picture.
![copq](https://cloud.githubusercontent.com/assets/24851355/22385383/625fcfec-e4a0-11e6-80a1-fc9bbefedfcf.jpeg)

**Comment by hluk** [comment]
Can try this with "Save and display HTML and rich text" option disabled? (It's under "Item" tab in Configuration dialog.)

![image](https://cloud.githubusercontent.com/assets/38054/22402509/f5bbb9cc-e5f9-11e6-80a3-11f85bc3dc14.png)

**Comment by briskycat** [comment]
I have this problem occasionally too, most often after suspend/resume. And I think I used to have it with FlyCut on earlier OS X releases.

**Comment by hluk** [comment]
It would be much easier to fix if anyone could debug the issue and send stack trace. I.e. build the app in debug mode and when the process starts to misbehave attach a debugger to it and print stacktrace (alternatively killing the process with SIGABRT might create core dump).

**Comment by briskycat** [comment]
I ran a built-in macOS profiler when it happened again:
[Sample of CopyQ.txt](https://github.com/hluk/CopyQ/files/747515/Sample.of.CopyQ.txt)
[Sample of CopyQ_2.txt](https://github.com/hluk/CopyQ/files/747518/Sample.of.CopyQ_2.txt)
Looks like the main culprit is PasteboardGetItemIdentifier.

**Comment by hluk** [comment]
Thanks for the info.

Seems that clipboard monitor process is actively looping in some Qt code when getting clipboard data.

Here is version with workaround (commit 6d399196b7ea4e6a4423d9c63a76725783c95e17): https://github.com/hluk/CopyQ/releases/download/testing/CopyQ.dmg

**Comment by flyinbutrs** [comment]
I have this same problem, and will also test that workaround.

**Comment by Fitech** [comment]
I was seeing the CPU hogging very frequently.  I'm using the workaround version and it hasn't happened since.  Thanks.

**Comment by briskycat** [comment]
I have been testing the workaround for more than a week now. And although in the log I have this:
```
Note [2017-02-02 23:53:03.741]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemdata.dylib
Note [2017-02-02 23:53:03.802]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemencrypted.dylib
Note [2017-02-02 23:53:03.826]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemfakevim.dylib
Note [2017-02-02 23:53:03.858]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemimage.dylib
Note [2017-02-02 23:53:03.894]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemnotes.dylib
Note [2017-02-02 23:53:03.938]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemsync.dylib
Note [2017-02-02 23:53:03.975]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemtags.dylib
Note [2017-02-02 23:53:04.008]: Loading plugin: /Applications/copyq.app/Contents/Plugins/copyq/libitemtext.dylib
ERROR [2017-02-03 13:06:56.811]: Remote process: Connection timeout!
ERROR [2017-02-07 12:02:56.115]: Remote process: Connection timeout!
ERROR [2017-02-13 11:50:16.076]: Remote process: Connection timeout!
ERROR [2017-02-14 16:15:42.970]: Remote process: Connection timeout!
ERROR [2017-02-14 16:16:00.970]: Remote process: Connection timeout!
ERROR [2017-02-14 16:32:11.946]: Remote process: Connection timeout!
ERROR [2017-02-14 16:32:28.946]: Remote process: Connection timeout!
ERROR [2017-02-14 16:32:45.945]: Remote process: Connection timeout!
ERROR [2017-02-14 16:33:02.945]: Remote process: Connection timeout!
ERROR [2017-02-14 16:33:19.944]: Remote process: Connection timeout!
ERROR [2017-02-14 16:33:35.944]: Remote process: Connection timeout!
ERROR [2017-02-14 16:33:53.944]: Remote process: Connection timeout!
ERROR [2017-02-14 16:34:09.943]: Remote process: Connection timeout!
ERROR [2017-02-14 16:34:25.945]: Remote process: Connection timeout!
```
I haven't observed any runaway processes ever since. So, I presume the workaround does work.

**Comment by hluk** [comment]
@briskycat Yup, the errors indicate that the clipboard monitor process isn't responding and it's restarted. Though it's strange that the process is restarted so many times.

**Comment by flyinbutrs** [comment]
I also haven't seen any runaway procs since testing started.

**Comment by briskycat** [comment]
@hluk It would be nice to reflect it in the log when the monitor process gets killed.

**Comment by hluk** [comment]
@briskycat I improved the error messages a bit (it will show something like `Restarting clipboard monitor (Connection timeout!)`).

**Comment by mrjcleaver** [comment]
Quick point of clarity - is this included in 3.0? The branch link above was updated 2 days ago, but 3.0 was released 23 days ago.

Thx.

**Comment by hluk** [comment]
@mrjcleaver The fix is included in 3.0. Is this still an issue?

**Comment by mrjcleaver** [comment]
Ah, no, it's seems fine now. Thanks!

**Comment by jialongl** [comment]
This bug is appearing again in the latest OS X test release after 3.0.1.

**Comment by hluk** [comment]
@jialongl I cannot reproduce it with latest [pre-release](https://github.com/hluk/CopyQ/releases/tag/osx-test) on Sierra 10.12.5. Do you know when it happens?

**Comment by jialongl** [comment]
@hluk I am sorry that I confused myself. The bug happened in version 3.0.1 actually, when I have copyQ main window shown. As soon as the main window shown, CPU usage hikes.

I used the "CopyQ About" window for the version info, and I thought the version number for the pre-release stays the same as the prior stable release.
I am now using the pre-release on Sierra 10.12.5, version 3.0.1-g22be297, and the problem is gone.

Sorry for all the confusion.
But by the way, you do have come across with the problem on 3.0.1, haven't you?

**Comment by hluk** [comment]
@jialongl Yes, I remember your issue was related to some rendering problems which was fixed some time after 3.0.1 was released.

The versioning for pre-release was also fixed recently.

I'll try to release new app version ASAP. Thanks for the report.

**Comment by jialongl** [comment]
No, thank you Lucas, for keeping up the good work!
